### PR TITLE
Remove duplicated types

### PR DESCRIPTION
**Pull request by rbehjati**
This PR unifies the two versions of `ValidatedProvenance` and moves the common part to a new package `types`. 

@mariaschett Let me know what you think.

### CODE REVIEW

File: pkg/types/types.go [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+		return nil, fmt.Errorf("the provenance must have exactly one subject with a sha256 digest")
+	}
+
+	// statement.Predicate is now just a map, we have to parse it into an instance of slsa.ProvenancePredicate
+	predicateBytes, err := json.Marshal(statement.Predicate)
+	if err != nil {
+		return nil, fmt.Errorf("could not marshal Predicate map into JSON bytes: %v", err)
+	}
+
+	var predicate slsa.ProvenancePredicate
```
Comment (mariaschett) [inline pkg/types/types.go:95]: I think, the downside is: at this point you are locked into slsav02.ProvenancePredicate. 

How will this change when there is a slsa v1.0?

**Comment by mariaschett** [comment]
Related to this the more general question: what do we guarantee for a `ValidatedProvenance`?

For know, I see the following guarantees:
- we can get a BinarySHA256Digest
- we can get a BinaryName.
- we can get a BuildType (maybe some guarantees on BuildType, too?).

I am also not sure, how `ValidatedProvenance` and `ProvenanceIR` relate to each other. I think they should.

Compare also #166 , where we were discussing a `ValidatedProvenanceValues`.

File: pkg/types/types.go (hunk 1 above)
Comment (rbehjati) [reply pkg/types/types.go:95]: This should not change for slsa v1.0 I think. But we have to wait and see when the new format is out. 
Note that the files in `slsa_provenance/v02` are copied from the `slsa` repo, but we have full control on the structs. If needed, we can update these to a new common denominator.

**Comment by rbehjati** [comment]
> Related to this the more general question: what do we guarantee for a `ValidatedProvenance`?
> 
> For know, I see the following guarantees:
> 
> * we can get a BinarySHA256Digest
> * we can get a BinaryName.
> * we can get a BuildType (maybe some guarantees on BuildType, too?).

I think that is about it. 

> I am also not sure, how `ValidatedProvenance` and `ProvenanceIR` relate to each other. I think they should.

We could potentially merge them into one. 

> Compare also #166 , where we were discussing a `ValidatedProvenanceValues`.

How does `ValidatedProvenanceValues` relate to `ProvenanceIR` now? Are not they going to be merged?

File: pkg/types/types.go (hunk 1 above)
Comment (rbehjati) [reply pkg/types/types.go:95]: Here is the slsa v1 draf: https://github.com/slsa-framework/slsa/pull/525/files. Let's discuss this part separately.

**Review by mariaschett — APPROVED** [review]
As discussed, LGTM!

**Comment by rbehjati** [comment]
> As discussed, LGTM!

Thanks. I made the changes we discussed :)

**Pull request merged**